src/store/modules/job/actions.ts

```diff
@@ -412,10 +418,14 @@ const actions: ActionTree<JobState, RootState> = {
         'runAsUser': 'system', //default system, but empty in run now.  TODO Need to remove this as we are using SERVICE_RUN_AS_SYSTEM, currently kept it for backward compatibility
         'recurrenceTimeZone': this.state.user.current.userTimeZone
       },
-      'shopifyConfigId': this.state.user.currentShopifyConfigId,
       'statusId': "SERVICE_PENDING",
       'systemJobEnumId': job.systemJobEnumId
     } as any
+    
+    if(job?.runtimeData?.shopifyConfigId) {
+      payload['shopifyConfigId'] = this.state.user.currentShopifyConfig?.shopifyConfigId
```

Comment:
Store `this.state.user.currentShopifyConfig` in a variable and check for it's value in `if` condition itself and remove optional chaining.